int256.go

```diff
@@ -10,8 +10,8 @@ var one = uint256.NewInt(1)
 var maxUint256 = uint256.MustFromHex("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
 
 type Int struct {
-	abs *uint256.Int
-	neg bool
+	Abs *uint256.Int
```

Comment:
Why did we put it public?
We don't want to leak and access these components from outside or other places that will use this package.

Reply:
Allowing the Int struct fields to be exported enables serialization with packages such as Gob, JSON, XML, etc. If the fields aren’t exported, serialization will result in an empty struct because these packages cannot access unexported fields.

I believe that having exported fields isn't an issue since their behavior is straightforward. In the uint256 library, the four uint64 fields that make up the type are also exported and can be modified outside of the package.

```
// Int is represented as an array of 4 uint64, in little-endian order,
// so that Int[3] is the most significant, and Int[0] is the least significant
type Int [4]uint64
```

Reply:
Hi @louisgthier 
Can you give me an example for this case?
`Allowing the Int struct fields to be exported enables serialization with packages such as Gob, JSON, XML, etc`

Reply:
> Allowing the Int struct fields to be exported enables serialization with packages such as Gob, JSON, XML, etc. If the fields aren’t exported, serialization will result in an empty struct because these packages cannot access unexported fields.
> 
> I believe that having exported fields isn't an issue since their behavior is straightforward. In the uint256 library, the four uint64 fields that make up the type are also exported and can be modified outside of the package.
> 
> ```
> // Int is represented as an array of 4 uint64, in little-endian order,
> // so that Int[3] is the most significant, and Int[0] is the least significant
> type Int [4]uint64
> ```
Regarding the Json serialization: 
- To allow a struct to be serialized to JSON or unserialized from JSON in Go, you generally need to create methods that implement the Marshaler and Unmarshaler interfaces defined in the encoding/json package. So no need to expose those fields, the implemention of those interfaces is enough. 

Almost sure something simillar exit for other XML, Gob etc format but I'm not familiar with those.

- "big.Int" have a struct similar to this package and those fields aren't exported: https://go.dev/src/math/big/int.go
- To marshal "big.int" to Json, they implement the interfaces needed for that here: https://go.dev/src/math/big/intmarsh.go